app/presenters/outcome_presenter.rb

```diff
@@ -1,7 +1,25 @@
 require 'erubis'
 
 class OutcomePresenter < NodePresenter
-  class OutcomeTemplateMissing < StandardError; end
+  class ViewContext
+    def initialize(state)
+      @state = state
+      state_methods = state.methods - SmartAnswer::State.new(nil).methods
+      @state_reader_methods = state_methods.reject { |m| m =~ /=$/ }
```

Comment:
I like `state_methods.reject { |m| m.to_s.end_with?('=') }` as it's more explicit than `state_methods.reject { |m| m =~ /=$/ }`. But again, it's a minor thing, a matter of taste rather than a readability concern :)